### PR TITLE
drivers: pinmux: 👋

### PR DESCRIPTION
**Pull request by gmarull**
Removal of pinmux is scheduled for 3.4 release (see https://github.com/zephyrproject-rtos/zephyr/issues/39740). Let's finally remove all pinmux drivers.

NOTE: many SoCs had custom pinmux implementations (e.g. `-pin` properties for Nordic). I'll work later on removing such legacy code.

### CODE REVIEW

**Review by simonguinot — DISMISSED** [review]
OK for the removal of the pinmux_lpc11u6x driver.

**Review by fabiobaltieri — DISMISSED** [review]

**Review by simonguinot — DISMISSED** [review]

**Review by erwango — DISMISSED** [review]

**Review by galak — DISMISSED** [review]

**Review by stephanosio — DISMISSED** [review]

**Comment by nashif** [comment]
let's give driver/soc/board maintainers/owners a couple of days to review and aknowledge before merging this.

**Comment by gmarull** [comment]
rebased to fix conflicts

**Review by fabiobaltieri — APPROVED** [review]

**Review by simonguinot — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by albertofloyd — APPROVED** [review]

**Pull request merged**
